bridges/snowbridge/pallets/inbound-queue-v2/src/test.rs

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+use super::*;
+
+use crate::{mock::*, Error};
+use frame_support::{assert_noop, assert_ok};
+use snowbridge_core::{inbound::Proof, sibling_sovereign_account};
+use sp_keyring::sr25519::Keyring;
+use sp_runtime::DispatchError;
+
+#[test]
+fn test_submit_with_invalid_gateway() {
+	new_tester().execute_with(|| {
+		let relayer: AccountId = Keyring::Bob.into();
+		let origin = RuntimeOrigin::signed(relayer);
+
+		// Submit message
+		let message = Message {
+			event_log: mock_event_log_invalid_gateway(),
+			proof: Proof {
+				receipt_proof: Default::default(),
+				execution_proof: mock_execution_proof(),
+			},
+		};
+		assert_noop!(
+			InboundQueue::submit(origin.clone(), message.clone()),
+			Error::<Test>::InvalidGateway
+		);
+	});
+}
+
+#[test]
+fn test_submit_happy_path() {
```

Comment:
maybe add another attempt here after the successful one and verify nonce verification fails

Reply:
Added a test: https://github.com/vgeddes/polkadot-sdk/blob/snowbridge-v2/bridges/snowbridge/pallets/inbound-queue-v2/src/test.rs#L120